### PR TITLE
Document citus backup and restore

### LINKED ISSUE #6887: Document Citus backup and restore
### Problem

While we wait on official backup and restore support from Stackgres, we should document the approach we take now.

### Solution

* Add a `docs/database/citus.md`
* Add a `Backup` section and document the steps
* Add a `Restore` section and document the steps

### Alternatives

_No response_

### PR DESCRIPTION
**Pull request by xin-hedera**
**Description**:

This PR adds a markdown document for citus backup and restore.

**Related issue(s)**:

Fixes #6887 

**Notes for reviewer**:

**Checklist**

- [x] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

File: docs/database/citus.md [hunk 1]
```diff
@@ -0,0 +1,155 @@
+# Citus Backup and Restore
```
Comment (steven-sheehy) [inline docs/database/citus.md:1]: Add a link to this doc from `docs/database/README.md` since that's the top-level doc.

**Review by steven-sheehy — APPROVED** [review]
LGTM

**Review by jnels124 — APPROVED** [review]
Looks good. I will mention that in the future may be worth exploring doing this work through the CRDs defined by zfs-localpv

**Comment by xin-hedera** [comment]
> Looks good. I will mention that in the future may be worth exploring doing this work through the CRDs defined by zfs-localpv

That can be a future item especially if zfs snapshot plus a small number of WAL segments is faster and needs less storage than backup & restore by stackgres.

**Pull request merged**